File: src/TemperatureDisplay.ts
import { ObserverInterface } from "./ObserverInterface";
import { WetherSubject } from "./WetherSubject";

export class TemperatureDisplay implements ObserverInterface {
  private temperature: number;

  constructor(wetherSubject: WetherSubject) {
    this.temperature = 0;
    wetherSubject.registerObserver(this);
  }

  update(temperature: number): void {
    this.temperature = temperature;
    this.display();
  }

  display(): void {
    console.log(`Current temperature: ${this.temperature}`);
  }
}

File: src/TemperatureDisplay.dart
import 'ObserverInterface.dart';
import 'WetherSubject.dart';

class TemperatureDisplay implements ObserverInterface {
  late double _temperature;

  TemperatureDisplay(WetherSubject wetherSubject) {
    _temperature = 0.0;
    wetherSubject.registerObserver(this);
  }

  void update(double temperature) {
    _temperature = temperature;
    display();
  }

  void display() {
    print("Current temperature: $_temperature");
  }
}
